src/mapping/impl/BasisFunctions.hpp

```diff
@@ -323,14 +517,32 @@ class CompactPolynomialC4 : public CompactSupportBase,
 
   double evaluate(double radius) const
   {
-    double p = radius * _r_inv;
-    if (p >= 1)
+    return this->operator()(radius, this->_params);
+  }
+
+  PRECICE_HOST_DEVICE inline double operator()(const double radius, const std::array<double, 3> &params) const
+  {
+    double       r_inv = params.at(0);
+    double const p     = radius * r_inv;
+    if (p >= 1) {
       return 0.0;
-    return math::pow_int<6>(1.0 - p) * (35 * math::pow_int<2>(p) + 18 * p + 3);
+    } else {
+      return std::pow(1.0 - p, 6) * (35 * std::pow(p, 2) + 18 * p + 3, 2);
```

Comment:
There is a weird `,2` at the end. `pow_int` is missing as well.

Reply:
Ah, this is probably what the compiler wants to tell me. But why does this compile anyways?